### PR TITLE
Reports v2

### PR DESCRIPTION
**Pull request by Bars92**
- Added List, Number & Trend.
- Added page size support.
- Added support for stacked bar chart.
- Fixes for line chart, pie chart, and bar chart. 
- Added title & titleStyle for all elements.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,10 +51,12 @@
     "postcss-reporter": "1.4.1",
     "react": "15.6.1",
     "react-dom": "15.6.1",
+    "react-grid-layout": "^0.16.2",
```
Comment (dorsha) [inline package.json]: remove the ^

File: reportsServer.js [hunk 2]
```diff
@@ -39,6 +59,7 @@ const resourceTimeout = system.args[5];
 const reportType = system.args[6] || 'pdf';
 const headerLeftImage = system.args[7] || '';
 const headerRightImage = system.args[8] || '';
+const pageSize = system.args[9] || PAGE_SIZES.A4;
```
Comment (dorsha) [inline reportsServer.js]: keep letter as default

File: src/components/Layouts/ReportLayout.js [hunk 3]
```diff
@@ -1,15 +1,168 @@
...
+import { isNumber } from 'lodash';
+import ReactGridLayout from 'react-grid-layout';
+const ROW_PIXEL_HEIGHT = 110;
+let overflowRows = 0;
 
-const ReportLayout = ({ sections, headerLeftImage, headerRightImage }) => {
+function getGridItemFromSection(section) {
+  const rows = section.layout.rowPos + overflowRows;
+  let height = section.layout.h;
+  if (section.type === SECTION_TYPES.table && section.layout.w === 12) {
```
Comment (dorsha) [inline src/components/Layouts/ReportLayout.js]: what is 12?

File: src/components/Sections/SectionList.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
+import React, { PropTypes } from 'react';
```
Comment (dorsha) [inline src/components/Sections/SectionList.js]: isn't this a table with two columns?

Comment (Bars92) [reply src/components/Sections/SectionList.js]: Yes, implemented the same but used as a list.

**Comment by Bars92** [comment]
image

**Comment by orius123** [comment]
Looks gr8

**Comment by dorsha** [comment]
@Slav4ik @sivanisentinel - please see above screenshot, we added new capabilities to our reports and fixed some issues.
We would like to share them with you and get your feedback.
You might want to use some of the new widgets.

**Pull request merged**